sentry_sdk/integrations/wsgi.py

```diff
@@ -78,17 +78,22 @@ def __call__(self, environ, start_response):
         hub = Hub(Hub.current)
 
         with hub:
+            span = None
             with capture_internal_exceptions():
                 with hub.configure_scope() as scope:
                     scope.clear_breadcrumbs()
                     scope._name = "wsgi"
-                    scope.set_span_context(SpanContext.continue_from_environ(environ))
                     scope.add_event_processor(_make_wsgi_event_processor(environ))
+                    scope.span = span = Span.continue_from_environ(environ)
 
             try:
                 rv = self.app(environ, start_response)
             except BaseException:
                 reraise(*_capture_exception(hub))
+            finally:
+                if span is not None:
+                    span.finish()
+                    hub.capture_trace(span)
```

Comment:
Could `capture_trace` look at the current trace in the scope, finish it and flush it?
Consider:
```python
with hub.trace(Span.continue_from_environ(..)):
   ..
```

Reply:
I rewrote this to use `with hub.trace` so any capture_trace-related changes no longer apply